### PR TITLE
test: update socket_passing.py to python3

### PR DESCRIPTION
**Pull request by keith**
This script is used for the hotrestart_test target, this makes it run
with python3 instead of python2

Fixes https://github.com/envoyproxy/envoy/issues/7379

Signed-off-by: Keith Smiley <keithbsmiley@gmail.com>

### CODE REVIEW

**Comment by keith** [comment]
I ran hotrestart_test on linux with this fix and this _seemed_ to be all that needed to change to support python3 in this script, but there could be more depending on the codepath that test hits. This would replace https://github.com/envoyproxy/envoy/pull/7382

File: tools/BUILD [hunk 1]
```diff
@@ -23,6 +23,7 @@ envoy_py_test_binary(
     srcs = [
         "socket_passing.py",
     ],
+    python_version = "PY3",
```
Comment (keith) [inline tools/BUILD]: We need this until we switch to bazel 0.27.0 on CI which changes this default

**Comment by htuch** [comment]
@keith these look like legit CI failures, want to take a look?

**Comment by keith** [comment]
It looks like it's still running with python2 somehow. I assume this would be fixed with the bazel update, I will try and debug more tomorrow

**Comment by htuch** [comment]
Why not just rebuild the CI image to pick up latest Bazel?

**Comment by keith** [comment]
Is this the right way to do that? https://github.com/envoyproxy/envoy/pull/7391

**Comment by htuch** [comment]
@keith yes, once we have #7391 we should be good here.

**Comment by keith** [comment]
https://github.com/envoyproxy/envoy/pull/7391

**Pull request closed without merging**